wandb/jupyter.py

```diff
@@ -97,12 +118,17 @@ def notebook_metadata(silent):
         # In colab we can request the most recent contents
         ipynb = attempt_colab_load_ipynb()
         if ipynb:
-            return {
+            ret = {
                 "root": "/content",
                 "path": ipynb["metadata"]["colab"]["name"],
                 "name": ipynb["metadata"]["colab"]["name"],
             }
 
+            jupyter_metadata = notebook_metadata_from_jupyter_servers_and_kernel_id()
+            if jupyter_metadata:
+                ret["path"] = jupyter_metadata["path"]
+                return ret
```

Comment:
Do we want this return outside of the `if jupyter_metadata` block?  What happens with `ret` if we couldn't talk to the metadata server in the colab case?

Reply:
Nice catch, thanks. I updated this to try getting `jupyter_metadata`, passing if it raises an error that it couldn't communicate with the server, updating path if it didn't, and returning `ret` in all cases. This also fixes the CI.